### PR TITLE
CA1054: CA1055: CA1056: Allow excluded_symbol_names user option

### PR DESCRIPTION
**Pull request by Evangelink**
## Summary

Update documentation following code update (see https://github.com/dotnet/roslyn-analyzers/pull/6058)

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca1055.md [hunk 1]
```diff
@@ -72,11 +72,14 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 Use the following option to configure which parts of your codebase to run this rule on.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1055.md:72]:
```suggestion
Use the following options to configure which parts of your codebase to run this rule on.
```

**Review by gewarren — COMMENTED** [review]
Thanks @Evangelink. I think we also need to list these three rules here: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-quality-rule-options#excluded_type_names_with_derived_types

**Comment by Evangelink** [comment]
@gewarren PR on roslyn-analyzers was merged so feel free to merge this when you want.

File: docs/fundamentals/code-analysis/code-quality-rule-options.md [hunk 2]
```diff
@@ -162,7 +162,7 @@ This section lists some of the available options. To see the full list of availa
...
...ca2301.md) [CA2302](quality-rules/ca2302.md) [CA2311](quality-rules/ca2311.md) [CA2312](quality-rules/ca2312.md)<br/>[CA2321](quality-rules/ca2321.md) [CA2322](quality-rules/ca2322.md) [CA2327](quality-rules/ca2327.md) [CA2328](quality-rules/ca2328.md)<br/>[CA2329](quality-rules/ca2329.md) [CA2330](quality-rules/ca2330.md) [CA3001](quality-rules/ca3001.md) [CA3002](quality-rules/ca3002.md)<br/>[CA3003](quality-rules/ca3003.md) [CA3004](quality-rules/ca3004.md) [CA3005](quality-rules/ca3005.md) [CA3006](quality-rules/ca3006.md)<br/>[CA3007](quality-rules/ca3007.md) [CA3008](quality-rules/ca3008.md) [CA3009](quality-rules/ca3009.md) [CA3010](quality-rules/ca3010.md)<br/>[CA3011](quality-rules/ca3011.md) [CA3012](quality-rules/ca3012.md) [CA5361](quality-rules/ca5361.md) [CA5376](quality-rules/ca5376.md)<br/>[CA5377](quality-rules/ca5377.md) [CA5378](quality-rules/ca5378.md) [CA5380](quality-rules/ca5380.md) [CA5381](quality-rules/ca5381.md)<br/>[CA5382](quality-rules/ca5382.md) [CA5383](quality-rules/ca5383.md) [CA5384](quality-rules/ca5384.md) [CA5387](quality-rules/ca5387.md)<br/>[CA5388](quality-rules/ca5388.md) [CA5389](quality-rules/ca5389.md) [CA5390](quality-rules/ca5390.md) |
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/code-quality-rule-options.md:165]: @Evangelink Did you mean to add them to excluded_symbol_names or excluded_type_names_with_derived_types?

Comment (Evangelink) [reply docs/fundamentals/code-analysis/code-quality-rule-options.md:165]: `excluded_symbol_names`.

File: docs/fundamentals/code-analysis/quality-rules/ca1054.md [hunk 3]
```diff
@@ -69,14 +69,17 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
...
 
 - [Include specific API surfaces](#include-specific-api-surfaces)
+- [Exclude specific types and their derived types](#exclude-specific-types-and-their-derived-types)
 
-You can configure this option for just this rule, for all rules, or for all rules in this category ([Design](design-warnings.md)). For more information, see [Code quality rule configuration options](../code-quality-rule-options.md).
+You can configure these options for just this rule, for all rules, or for all rules in this category ([Design](design-warnings.md)). For more information, see [Code quality rule configuration options](../code-quality-rule-options.md).
 
 [!INCLUDE[api-surface](~/includes/code-analysis/api-surface.md)]
 
+[!INCLUDE[excluded-type-names-with-derived-types](~/includes/code-analysis/excluded-type-names-with-derived-types.md)]
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1054.md:81]:
```suggestion
[!INCLUDE[excluded-symbol-names](~/includes/code-analysis/excluded-symbol-names.md)]
```

File: docs/fundamentals/code-analysis/quality-rules/ca1055.md [hunk 4]
```diff
@@ -69,14 +69,17 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
...
 
 - [Include specific API surfaces](#include-specific-api-surfaces)
+- [Exclude specific types and their derived types](#exclude-specific-types-and-their-derived-types)
 
-You can configure this option for just this rule, for all rules, or for all rules in this category ([Design](design-warnings.md)). For more information, see [Code quality rule configuration options](../code-quality-rule-options.md).
+You can configure these options for just this rule, for all rules, or for all rules in this category ([Design](design-warnings.md)). For more information, see [Code quality rule configuration options](../code-quality-rule-options.md).
 
 [!INCLUDE[api-surface](~/includes/code-analysis/api-surface.md)]
 
+[!INCLUDE[excluded-type-names-with-derived-types](~/includes/code-analysis/excluded-type-names-with-derived-types.md)]
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1055.md:81]:
```suggestion
[!INCLUDE[excluded-symbol-names](~/includes/code-analysis/excluded-symbol-names.md)]
```

File: docs/fundamentals/code-analysis/quality-rules/ca1056.md [hunk 5]
```diff
@@ -69,14 +69,17 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
...
 
 - [Include specific API surfaces](#include-specific-api-surfaces)
+- [Exclude specific types and their derived types](#exclude-specific-types-and-their-derived-types)
 
-You can configure this option for just this rule, for all rules, or for all rules in this category ([Design](design-warnings.md)). For more information, see [Code quality rule configuration options](../code-quality-rule-options.md).
+You can configure these options for just this rule, for all rules, or for all rules in this category ([Design](design-warnings.md)). For more information, see [Code quality rule configuration options](../code-quality-rule-options.md).
 
 [!INCLUDE[api-surface](~/includes/code-analysis/api-surface.md)]
 
+[!INCLUDE[excluded-type-names-with-derived-types](~/includes/code-analysis/excluded-type-names-with-derived-types.md)]
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1056.md:81]:
```suggestion
[!INCLUDE[excluded-symbol-names](~/includes/code-analysis/excluded-symbol-names.md)]
```

File: docs/fundamentals/code-analysis/quality-rules/ca1054.md [hunk 6]
```diff
@@ -69,14 +69,17 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 ## Configure code to analyze
 
-Use the following option to configure which parts of your codebase to run this rule on.
+Use the following options to configure which parts of your codebase to run this rule on.
 
 - [Include specific API surfaces](#include-specific-api-surfaces)
+- [Exclude specific types and their derived types](#exclude-specific-types-and-their-derived-types)
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1054.md:75]:
```suggestion
- [Exclude specific symbols](#exclude-specific-symbols)
```

File: docs/fundamentals/code-analysis/quality-rules/ca1055.md [hunk 7]
```diff
@@ -69,14 +69,17 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 ## Configure code to analyze
 
-Use the following option to configure which parts of your codebase to run this rule on.
+Use the following options to configure which parts of your codebase to run this rule on.
 
 - [Include specific API surfaces](#include-specific-api-surfaces)
+- [Exclude specific types and their derived types](#exclude-specific-types-and-their-derived-types)
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1055.md:75]:
```suggestion
- [Exclude specific symbols](#exclude-specific-symbols)
```

File: docs/fundamentals/code-analysis/quality-rules/ca1056.md [hunk 8]
```diff
@@ -69,14 +69,17 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 ## Configure code to analyze
 
-Use the following option to configure which parts of your codebase to run this rule on.
+Use the following options to configure which parts of your codebase to run this rule on.
 
 - [Include specific API surfaces](#include-specific-api-surfaces)
+- [Exclude specific types and their derived types](#exclude-specific-types-and-their-derived-types)
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1056.md:75]:
```suggestion
- [Exclude specific symbols](#exclude-specific-symbols)
```

**Review by gewarren — COMMENTED** [review]
In that case I believe the include files were wrong.

**Comment by Evangelink** [comment]
Sorry it's been a while since I touched this user option. I have been checking the code and the method we use actually always checks for both. So the list of supported rules should always be equal for the 2 options.

File: docs/fundamentals/code-analysis/code-quality-rule-options.md [hunk 9]
```diff
@@ -156,13 +156,13 @@ This section lists some of the available options. To see the full list of availa
...
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/code-quality-rule-options.md:159]: I'll log an issue to add the include file to the individual rule pages for the rules that have been added here.

**Review by gewarren — APPROVED** [review]

**Pull request merged**